src/components/sections/finding/MyFindingDetails.tsx

```diff
@@ -0,0 +1,197 @@
+
+import {Button} from '@/components/ui/Button'
+import Separator from '@/components/ui/Separator'
+import Skeleton from '@/components/ui/Skeleton'
+import {formatAda, formatDate} from '@/lib/utils/common/format'
+import {getContestStatus} from '@/lib/utils/common/contest'
+import {useGetMyFinding} from '@/lib/queries/finding/getMyFinding'
+import {PATHS} from '@/lib/utils/common/paths'
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from '@/components/ui/AlertDialog'
+import {useDeleteFinding} from '@/lib/queries/finding/deleteFinding'
+
+type MyFindingDetailsProps = {
+  findingId: string
+}
+
+const MyFindingDetails = ({findingId}: MyFindingDetailsProps) => {
+  const {data, isLoading} = useGetMyFinding({findingId})
+  const {mutateAsync, isPending} = useDeleteFinding()
+  const router = useRouter()
+
+  if (isLoading || isPending) {
+    return (
+      <div className="px-24">
+        <div className="mb-12">
+          <Skeleton className="h-[72px]" />
+        </div>
+        <Separator />
+        <div className="mt-12 flex flex-col gap-3">
+          <Skeleton className="h-[72px]" />
+          <Skeleton className="h-[72px]" />
+          <Skeleton className="h-[72px]" />
+        </div>
+      </div>
+    )
+  }
+
+  if (!data) {
+    return notFound()
+  }
+
+  return (
+    <>
+      <div className="flex h-[136px] items-center gap-12 px-24">
+        <Button
+          variant="outline"
+          size="small"
+          className="flex gap-2 uppercase"
+          onClick={() => router.back()}>
+          <ArrowLeft width={16} height={16} />
+          Go Back
+        </Button>
+      </div>
+
+      <div className="flex flex-grow flex-col bg-black px-24 pb-24 pt-12 xl:px-[340px]">
+        <div className="mb-12 flex flex-col gap-3">
+          <h3 className="text-bodyL text-purple-light">Your wallet address</h3>
+          <span className="text-bodyM">{data.rewardWalletAddress}</span>
+        </div>
+
+        <div className="mb-12 flex gap-12">
+          <div className="flex flex-col">
+            <span className="text-bodyL text-purple-light">Project name</span>
+            <span className="text-bodyM">{data.contest.title}</span>
+          </div>
+          <div className="flex flex-col">
+            <span className="text-bodyL text-purple-light">Submitted</span>
+            <span className="text-bodyM">
+              {formatDate(data.createdAt, DateTime.DATETIME_MED)}
+            </span>
+          </div>
+          <div className="flex flex-col">
+            <span className="text-bodyL text-purple-light">Project state</span>
+            <span className="text-bodyM capitalize">
+              {getContestStatus({
+                startDate: data.contest.startDate,
+                endDate: data.contest.endDate,
+                status: data.contest.status,
+              })}
```

Comment:
Hmm.. I don't think I understand what you mean by decoupling to a mapping. Currently there is already a `translateEnum` translation for `ContestStatus`. Do you mean to use that instead? I checked the other usages of `getContestStatus` and those seem appropriate because even if the function is used in displaying findings the `inReview` and `notApproved` states will never show anyway.

Reply:
I mean separating the status from value that's used to display as a text, for example by having a mapping like this:
```
const contestStatusTexts = {
  approved: 'Approved',
  inReview: 'In review',
  ...
}
```
instead of using the status directly.

Reply:
Ahh I see now.  I chose to organize this using enums to maintain consistency with the existing structure of the application.

I created an enum ContestStatusText:
```ts
export enum ContestStatusText {
  draft = 'draft',
  rejected = 'rejected',
  finished = 'finished',
  notApproved = 'notApproved',
  inReview = 'inReview',
  pending = 'pending',
  judging = 'judging',
  live = 'live',
  approved = 'approved',
}
```

This is then returned by the `getContestStatus` function (renamed to `getContestStatusText`) which then needs to be translated by the `translateEnum` util before showing to the client.